pallets/schemas/src/lib.rs

```diff
@@ -258,20 +259,20 @@ pub mod pallet {
 		/// Set the schema count to something in particular.
 		#[cfg(any(feature = "std", feature = "runtime-benchmarks", test))]
 		pub fn set_schema_count(n: SchemaId) {
-			<SchemaCount<T>>::set(n);
+			<CurrentSchemaIdentifierMaximum<T>>::set(n);
 		}
 
 		fn add_schema(
 			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
 			model_type: ModelType,
 			payload_location: PayloadLocation,
 		) -> Result<SchemaId, DispatchError> {
-			let cur_count = Self::get_schema_count();
+			let cur_count = Self::get_current_schema_identifier_maximum();
 			ensure!(cur_count < T::MaxSchemaRegistrations::get(), Error::<T>::TooManySchemas);
 			let schema_id = cur_count.checked_add(1).ok_or(Error::<T>::SchemaCountOverflow)?;
```

Comment:
Should we encapsulate this logic in a method similar to the MSA pallet?
```
	pub fn get_next_msa_id() -> Result<MessageSourceId, DispatchError> {
		let next = Self::get_identifier().checked_add(1).ok_or(Error::<T>::MsaIdOverflow)?;
		let next = Self::get_current_msa_identifier_maximum()
			.checked_add(1)
			.ok_or(Error::<T>::MsaIdOverflow)?;

		Ok(next)
	}

```

Reply:
I think this is different and safe way to do count+1

In the one you commented, that is getting the total count while adding which is bound to be the number current schema is being added again

Reply:
I see what you mean having looked at both functions


Good suggestion let me do it